### PR TITLE
Fixed get student groups enrolled error

### PR DESCRIPTION
**Pull request by stavliv**
## Summary
getStudents test raised error, groupsEnrolled didn't have the expected values and format

## Changes
* Fixed example in student schema for groups enrolled from str to array
* Made the change also to the student service
*  when comparing arrays in tests we need t.deepEqual() instead of t.is()
* the same changes were made to coach->groupsCreated **WARNING: not tested**
* deleted updateClassroom empty test

## Tests
* locally run the getStudents test

### CODE REVIEW

**Comment by stavliv** [comment]
We can decide whether we want the groupsEnrolled attribute to be a str or an array in the responses, if we want it as an array, accept this pr, otherwise we can make our tests to expect a str fromat

**Comment by stavliv** [comment]
> We can decide whether we want the groupsEnrolled attribute to be a str or an array in the responses, if we want it as an array, accept this pr, otherwise we can make our tests to expect a str fromat

based on this -> https://github.com/OAI/OpenAPI-Specification/blob/main/tests/v3.1/fail/invalid_schema_types.yaml maybe we should let them strings?

**Comment by isidorostsa** [comment]
> based on this -> https://github.com/OAI/OpenAPI-Specification/blob/main/tests/v3.1/fail/invalid_schema_types.yaml maybe we should let them strings?

@stavliv I thinks arrays should be fine, under [openapi's example specification](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#example-object) it says:
image

So I assume that since arrays can be naturally represented in json we should be ok. Also there is an example of a schema featuring an array [here](github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#paths-object-example)

**Comment by stavliv** [comment]
> > based on this -> https://github.com/OAI/OpenAPI-Specification/blob/main/tests/v3.1/fail/invalid_schema_types.yaml maybe we should let them strings?
> 
> @stavliv I thinks arrays should be fine, under [openapi's example specification](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#example-object) it says: image
> 
> So I assume that since arrays can be naturally represented in json we should be ok. Also there is an example of a schema featuring an array [here](github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#paths-object-example)

In that case the decesion falls on us, do we keep them as strings and make the tests to expect strings or modify the swagger generated code to send arrays?

File: tests/postGroupsEnroll.test.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    t.true(body.hasOwnProperty('groupID'));
+    t.true(body.hasOwnProperty('studentID'));
+
+    // check the values are correct
+    t.is(body.groupID, group_id);
+    t.is(body.studentID, student_id);
+});
+
+
+test('POST groups/enroll unsuccessful', async (t) => {
```
Comment (stavliv) [inline tests/postGroupsEnroll.test.js:91]: @isidorostsa @alexisbouletsis @KonstantinaGkavanozi  if anyone has found a cleaner way to test bad cases that would include checking specifically for the code 400, please contribute it :)

File: service/GroupService.js [hunk 2]
```diff
@@ -52,10 +52,11 @@ exports.deleteGroup = function(groupID) {
...
     var examples = {};
-    examples['application/json'] = {
-  "studentID" : 0,
-  "groupID" : 6
-};
+    // examples['application/json'] = {
+    //   "studentID" : 0,
+    //   "groupID" : 6
+    // };
+    examples['application/json'] = body;
```
Comment (stavliv) [inline service/GroupService.js:59]: This is to have some actuall logic on the server and not always return the example no matter how messed up the query (so the tests have some meaning).

**Comment by stavliv** [comment]
Closed. Going with Isidoros' approach.

**Pull request closed without merging**